CONTRIBUTING.md

```diff
@@ -0,0 +1,281 @@
+multiple permutations of ways to achieve that goal in-depth, and the end result of a tutorial does not need to
+be completely production-ready against all cases. Information on how to address additional scenarios
+can occur in how-to guides.
+
+To quote the Diataxis website:
+
+> A tutorial serves the user’s *acquisition* of skills and knowledge - their study. Its purpose is not to help the user get something done, but to help them learn.
+
+In LangGraph, these are often higher level guides that show off end-to-end use cases.
+
+Some examples include:
+
+- [Build a Customer Support Bot](https://langchain-ai.github.io/langgraph/tutorials/customer-support/customer-support/)
+- [Build a SQL Agent](https://langchain-ai.github.io/langgraph/tutorials/sql-agent/)
+
+Here are some high-level tips on writing a good tutorial:
+
+- Focus on guiding the user to get something done, but keep in mind the end-goal is more to impart principles than to create a perfect production system
+- Be specific, not abstract and follow one path
+  - No need to go deeply into alternative approaches, but it’s ok to reference them, ideally with a link to an appropriate how-to guide
+- Get "a point on the board" as soon as possible - something the user can run that outputs something
+  - You can iterate and expand afterwards
+  - Try to frequently checkpoint at given steps where the user can run code and see progress
+- Focus on results, not technical explanation
+  - Crosslink heavily to appropriate conceptual/reference pages
+- The first time you mention a LangGraph concept, use its full name (e.g. "human-in-the-loop"), and link to its conceptual/other documentation page
+  - It's also helpful to add a prerequisite callout that links to any pages with necessary background information
+- End with a recap/next steps section summarizing what the tutorial covered and future reading, such as related how-to guides
+
+### How-to guides
+
+A how-to guide, as the name implies, demonstrates how to do something discrete and specific.
+It should assume that the user is already familiar with underlying concepts, and is trying to solve an immediate problem, but
+should still give some background or list the scenarios where the information contained within can be relevant.
+They can and should discuss alternatives if one approach may be better than another in certain cases.
+
+To quote the Diataxis website:
+
+> A how-to guide serves the work of the already-competent user, whom you can assume to know what they want to do, and to be able to follow your instructions correctly.
+
+Some examples include:
+
+- [How to add persistence to your graph](https://langchain-ai.github.io/langgraph/how-tos/persistence/)
+- [How to view and update past graph state](https://langchain-ai.github.io/langgraph/how-tos/human_in_the_loop/time-travel/)
+
+Here are some high-level tips on writing a good how-to guide:
+
+- Clearly explain what you are guiding the user through at the start
```

Comment:
Should we add something like "prefer full code blocks that the reader can copy and run" since people often are hunting for a snippet that directly solves a problem they know they have?